pkg/segment/reader/segread/segreader/segreader.go

```diff
@@ -83,6 +112,57 @@ type SegmentFileReader struct {
 	allBmi             *structs.AllBlksMetaInfo
 }
 
+func GetBufFromPool(size int64) []byte {
+	switch {
+	case size <= S_1_KB:
+		return pool1K.Get(S_1_KB)[:size]
+	case size <= S_4_KB:
+		return pool4K.Get(S_4_KB)[:size]
+	case size <= S_32_KB:
+		return pool32K.Get(S_32_KB)[:size]
+	case size <= S_64_KB:
+		return pool64K.Get(S_64_KB)[:size]
+	case size <= S_128_KB:
+		return pool128K.Get(S_128_KB)[:size]
+	case size <= S_256_KB:
+		return pool256K.Get(S_256_KB)[:size]
+	case size <= S_512_KB:
+		return pool512K.Get(S_512_KB)[:size]
+	case size <= S_1_MB:
+		return pool1M.Get(S_1_MB)[:size]
+	case size <= S_2_MB:
+		return pool2M.Get(S_2_MB)[:size]
+	default: // TODO release
+		return make([]byte, 0, size) // too big, don't pool
+	}
+}
+
+func PutBufToPool(buf []byte) error {
+	switch cap(buf) {
+	case S_1_KB:
+		return pool1K.Put(buf)
+	case S_4_KB:
+		return pool4K.Put(buf)
+	case S_32_KB:
+		return pool32K.Put(buf)
+	case S_64_KB:
+		return pool64K.Put(buf)
+	case S_128_KB:
+		return pool128K.Put(buf)
+	case S_256_KB:
+		return pool256K.Put(buf)
+	case S_512_KB:
+		return pool512K.Put(buf)
+	case S_1_MB:
+		return pool1M.Put(buf)
+	case S_2_MB:
+		return pool2M.Put(buf)
+	default:
+		// TODO - log error??
```

Comment:
this is not an error, since we have been asked for a larger pool, we took from the pool by make cmd and while returning, we let GC take care of it